tests/src/test/scala/com/nvidia/spark/rapids/CachedBatchWriterSuite.scala

```diff
@@ -250,6 +242,8 @@ class CachedBatchWriterSuite extends SparkQueryCompareTestSuite {
       val cb = cachedBatchIter.next()
       totalRows += cb.numRows
     }
+    assertThrows[IllegalStateException](
+      cachedBatchIter.asInstanceOf[producer.ColumnarBatchToCachedBatchIterator].close())
```

Comment:
The close shouldn't throw per the discussion above about close being called twice under normal circumstances.  I think the line above that is:
```scala
  val cachedBatchIter = producer.getColumnarBatchToCachedBatchIterator
```
should be
```scala
  withResource(producer.getColumnarBatchToCachedBatchIterator) { cachedBatchIter =>
```

Reply:
I thought the suggestion was to close the iterator automatically when it is exhausted. No need to call close on it then

Reply:
If the test crashes (i.e.: one way test does not pass) then we would like to clean up, if possible.